File: php/password/hash.php
#!/usr/bin/env php
<?php
/**
 * @author stev leibelt <artodeto@bazzline.net>
 * @since 2016-06-29
 * @see
 *  http://php.net/manual/en/ref.password.php
 *  http://php.net/manual/en/function.password-hash.php
 */

if ($argc < 2) {
    echo 'invalid number of arguments provided' . PHP_EOL;
    echo basename(__FILE__) . ' <password>' . PHP_EOL;
    exit(1);
}

$password   = $argv[1];
$hash       = password_hash($password, PASSWORD_DEFAULT);

echo 'password: ' . $password . PHP_EOL;
echo 'hash: ' . $hash . PHP_EOL;

File: python.d/data_type.d/string.d/hash.py
#!/usr/bin/python
####
# @since 2025-01-14
# @author stev leibelt <artodeto@bazzline.net>
####

from hashlib import sha3_512

def main() -> None:
    my_string: str = "This is a string"

    my_hash: str = sha3_512(my_string.encode('utf-8')).hexdigest()

    print(f"{my_string=}: {str(my_hash)=}, {len(my_hash)=}")

if __name__ == '__main__':
    main()
